src/transformers/models/bart/modeling_bart.py

```diff
@@ -263,14 +287,229 @@ def forward(
         return attn_output, attn_weights_reshaped, past_key_value
 
 
+class BartFlashAttention2(BartAttention):
+    """
+    Bart flash attention module. This module inherits from `BartAttention` as the weights of the module stays
+    untouched. The only required change would be on the forward pass where it needs to correctly call the public API of
+    flash attention and deal with padding tokens in case the input contains any of them.
+    """
+
+    def _shape(self, tensor: torch.Tensor, seq_len: int, bsz: int):
```

Comment:
nit: this name isn't super informative and conflicts with `.shape`. Could we do `_reshape` instead?

Reply:
I'll add a copied from here! It's from `BartAttention`. Would probably be better to rename it everywhere at once

Reply:
Actually it's not the same -> yeah ok with renaming it to `reshape`